ql/math/randomnumbers/splitmix64.cpp

```diff
@@ -0,0 +1,25 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2023 Ralf Konrad Eckel
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/math/randomnumbers/seedgenerator.hpp>
+#include <ql/math/randomnumbers/splitmix64.hpp>
+
+namespace QuantLib {
+    SplitMix64::SplitMix64(uint64_t x) : x_(x) {}
```

Comment:
You can inline this in the hpp and do away with the cpp.

Reply:
Yes, was on my list as well. Was waiting if we keep it as a separate class or not.